### DISCUSSION TITLE
Is it a good idea to add a beforeRetry callback to the retry operator?

### DISCUSSION THREAD

**Discussion by tbontb-iaq** [ReactiveX/rxjs#7490] [Ideas / Feature request]
Add a `beforeRetry` property in `RetryConfig`:

```ts
interface RetryConfig {
  count?: number
  delay?: number | ((error: any, retryCount: number) => ObservableInput<any>)
  resetOnSuccess?: boolean
  beforeRetry?: (reason?: unknown) => void
}
```

Pass captured error to the callback before each retry, which can be used to record error info, etc.

**Comment by turien-pix4d** [comment]
I can think of two ways to do this : 
 1. Use the `delay` function of the `RetryConfig`, as it is guaranteed this function will be called for each error before retrying.
 2. Preeced the call of the retry operator with a `tap({error :  (reason: unknown) => {/* ... */}})`
